### PR TITLE
Change test-infra to kubernetes-sigs/prow

### PR DESCRIPTION
**Pull request by psalajova**
> This script checks that the vendored version of the test-infra repo is not newer than those vendored into the release controller and ci chat bot.

Since prow changed repositories from kubernetes/test-infra to kubernetes-sigs/prow, and we are updating our dependencies to work with this new import path, this script (used in the `pull-ci-openshift-ci-tools-master-validate-test-infra` presubmit job) should reflect this change as well. 

Should probably be merged after https://github.com/openshift/ci-tools/pull/4203 merges.

/cc @danilo-gemoli @jmguzik

### CODE REVIEW

File: hack/check-test-infra-version.sh [hunk 1]
```diff
@@ -29,14 +29,14 @@ ci_chat_bot_vendored_commit="$( determine_vendored_commit ./../ci-chat-bot )"
 
 failures=0
 
-pushd ./../../kubernetes/test-infra
+pushd ./../../kubernetes-sigs/prow
```
Comment (psalajova) [inline hack/check-test-infra-version.sh:32]: I'm not sure about this line..

Comment (danilo-gemoli) [reply hack/check-test-infra-version.sh:32]: It should be fine

**Comment by danilo-gemoli** [comment]
Can you also change the script filename from into `hack/check-prow-version.sh`?

**Pull request merged**